src/main/java/com/io/image/manager/service/operations/ImageOperationParser.java

```diff
@@ -0,0 +1,37 @@
+package com.io.image.manager.service.operations;
+
+import java.util.*;
+
+public class ImageOperationParser {
+
+    public static List<ImageOperation> parse(String query) {
+        List<ImageOperation> imageOperations = new ArrayList<>();
+
+        String[] params = query.split("&");
+
+        ImageOperation imageOperation = null;
+
+        for (String param : params) {
+            String[] args = param.split("=");
+
+            if ("op".equals(args[0])) {
```

Comment:
For a readability purposes please assign those `args[0]` and `args[1]` to something like `key` and `value` as we don't expect there to be more variables than that. And I think we may take advantage of some library to parse query params for us (provided that it does not change the order), then it could be more readable than now and this may be error prone.